zarr/util.py

```diff
@@ -312,117 +313,126 @@ def _repr_html_(self):
-        name += "[...]" if hasattr(node, "dtype") else ""
-        return name
-
+        return node.get_text()
 
-def custom_html_sublist(group, indent):
-    traverser = ZarrGroupTraversal(tree=group)
-    result = ""
 
-    result += (
-        """{0}<li><div>{1}</div>""".format(
-            indent, traverser.get_text(group)
-        )
-    )
-
-    children = traverser.get_children(group)
+def tree_html_sublist(node, root=False, expand=False):
+    result = ''
+    data_jstree = '{"type": "%s"}' % node.get_type()
+    css_class = 'jstree-open' if (root or expand) else ''
+    result += "<li data-jstree='{}' class='{}'>".format(data_jstree, css_class)
+    result += '<span>{}</span>'.format(node.get_text())
+    children = node.get_children()
     if children:
-        result += """\n{0}{0}<ul>\n""".format(indent)
+        result += '<ul>'
         for c in children:
-            for l in custom_html_sublist(c, indent).splitlines():
-                result += "{0}{0}{1}\n".format(indent, l)
-        result += "{0}{0}</ul>\n{0}".format(indent)
-
-    result += "</li>\n"
-
+            result += tree_html_sublist(c, expand=expand)
+        result += '</ul>'
+    result += '</li>'
     return result
 
 
-def custom_html_list(group, indent="    "):
-    result = ""
-
-    # Add custom CSS style for our HTML list
-    result += """<style type="text/css">\n"""
-    result += dedent("""\
-        div.zarr-tree {
-            font-family: Courier, monospace;
-            font-size: 11pt;
-            font-style: normal;
-        }
-
-        div.zarr-tree ul,
-        div.zarr-tree li,
-        div.zarr-tree li > div {
-            display: block;
-            position: relative;
-        }
-
-        div.zarr-tree ul,
-        div.zarr-tree li {
-            list-style-type: none;
-        }
-
-        div.zarr-tree li {
-            border-left: 2px solid #000;
-            margin-left: 1em;
-        }
-
-        div.zarr-tree li > div {
-            padding-left: 1.3em;
-            padding-top: 0.225em;
-            padding-bottom: 0.225em;
-        }
-
-        div.zarr-tree li > div::before {
-            content: '';
-            position: absolute;
-            top: 0;
-            left: -2px;
-            bottom: 50%;
-            width: 1.2em;
-            border-left: 2px solid #000;
-            border-bottom: 2px solid #000;
-        }
-
-        div.zarr-tree > ul > li:first-child > div {
-            padding-left: 4%;
-        }
-
-        div.zarr-tree > ul > li:first-child > div::before {
-            border: 0 none transparent;
-        }
-
-        div.zarr-tree ul > li:last-child {
-            border-left: 2px solid transparent;
-        }
-    """)
-    result += "</style>\n\n"
-
-    # Insert the HTML list
-    result += """<div class="zarr-tree">\n"""
-    result += "<ul>\n"
-    result += custom_html_sublist(group, indent=indent)
-    result += "</ul>\n"
-    result += "</div>\n"
+def tree_html(group, expand, level):
```

Comment:
This is probably my error in the first place, but should we just make this a method of `TreeViewer`?

Reply:
Yes, could do.